kubetest2/kubetest2-gke/deployer/commandutils.go

```diff
@@ -0,0 +1,159 @@
+package deployer
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"k8s.io/test-infra/kubetest2/pkg/exec"
+)
+
+func (d *deployer) prepareGcpIfNeeded() error {
```

Comment:
this code could do with some more refactoring and documenting, but I understand that this is mostly a transplant of functionality reverse engineered from kubetest :-)

Reply:
this is a lot better now